src/ScreenshotManager.vala

```diff
@@ -162,6 +178,44 @@ namespace Gala
 			selection_area.get_selection_rectangle (out x, out y, out width, out height);
 		}
 
+		private void unconceal_text ()
+		{
+			if (!text_concealed) {
+				return;
+			}
+
+			var desktop_settings = new Settings ("org.gnome.desktop.interface");
+
+			desktop_settings.set_string ("font-name", prev_font_regular);
+			desktop_settings.set_string ("monospace-font-name", prev_font_mono);
+			desktop_settings.set_string ("document-font-name", prev_font_document);
+			text_concealed = false;
+		}
+
+		public async void conceal_text () throws DBusError, IOError
+		{
+			if (text_concealed) {
+				return;
+			}
+
+			Timeout.add (UNCONCEAL_TEXT_TIMEOUT, () => {
```

Comment:
I'm not sure if the choice to ignore subsequent calls to `conceal_text` is correct here. I feel like instead, a subsequent call should cancel the current timeout and add a new one with a duration of 2s.